packages/ogm/src/classes/OGM.ts

```diff
@@ -23,27 +23,29 @@ import { filterDocument } from "../utils";
 
 export type OGMConstructor = Neo4jGraphQLConstructor;
 
-class OGM {
+class OGM<ModelMap = {}> {
     public models: Model[];
 
     checkNeo4jCompat: () => Promise<void>;
 
+    public neoSchema: Neo4jGraphQL;
+
     constructor(input: OGMConstructor) {
         const { typeDefs, ...rest } = input;
 
-        const neoSchema = new Neo4jGraphQL({
+        this.neoSchema = new Neo4jGraphQL({
             ...rest,
             typeDefs: filterDocument(typeDefs),
         });
 
         this.checkNeo4jCompat = function checkNeo4jCompat() {
-            return neoSchema.checkNeo4jCompat({
+            return this.neoSchema.checkNeo4jCompat({
```

Comment:
Now that `neoSchema` is a class parameter, couldn't we make `checkNeo4jCompat` a simple public function?

```ts
checkNeo4jCompat(): Promise<void>{
    return this.neoSchema.checkNeo4jCompat({
        drive: this.rest.driver,  // rest should be a private parameter
        ...(rest.config?.driverConfig ? { driverConfig: this.rest.config.driverConfig } : {})
       })
}
```

I think is simpler and more readable than a curryfied function in the constructor

Reply:
Good observation, I'm going to merge this seeming the pipelines passed for this and knowing making a change will leave another PR unmerged until it's fixed.